### ISSUE TITLE
Docs updates: Running To-Do list

### ISSUE THREAD

**Issue by rhine3** [kitzeslab/opensoundscape#335]
Minor clarity updates to documentation. Explanations inline

* Difference between `model.load()` and `Class.from_checkpoint()`: 
> The difference technically is that for .load() you have to create a model instance first then model.load() - this means you need to know the classes a priori. Whereas from checkpoint is a class method so you just call Class.from_checkpoint on a model path without knowing anything more than the Class the model was created with. Practically, .from_checkpoint is not implemented for the generic class because it needs to know what architecture to use - this is something we could add unless I’m forgetting another barrier

* Purpose of `Resnet18Binary` and `Resnet18Multiclass` classes; distinguish them as separate classes from the general PytorchModel framework. Possibly in the basic tutorial use PytorchModel instead of ResNet18... classes?
> So what happened is the ResNet18Binary and Multiclass are based on the original classes Miao wrote, but are a bit more “custom” than the most generic opso classes. The PytorchModel class is the generic class that can take any architecture, including resnet, and the Inception classes are that class adapted for inception. Those are the classes that fit the opso framework and structure, and we should maintain. The Resnet18… classes are kind of legacy or custom in that they use an internally implemented version of the resnet architecture. Thats why they don’t have the same initialization structure, and are created a bit differently. Maybe they should be renamed and tutorials should not use them. But they are what we’ve mostly been using to train models, because they allow the added flexibility of separate training parameters for feature and classification layers. (Enabling this generally would require us to have internal implantation of every architecture). [from #328]

* Note that can't view all training params by looking at the `optimizer_params` dict; see #330

Might add more as we find them.

**Comment by sammlapp** [comment]
Small addition for Custom Preprocessing notebook:
after showing how to create a custom Action, show how to use it with a CNN. Eg, if you created a custom Action class called SpecToImgColormap:

#add or overwrite an action to the dataset
train_dataset.actions.to_img_color = SpecToImgColormap()
# add the action to the appropriate location in the pipeline
train_dataset.pipeline[4]=train_dataset.actions.to_img_color

**Comment by rhine3** [comment]
Ideas from @lydiakatsis:

* Include examples of how to plot additional metrics apart from loss, for both validation and training data
* How to cite

**Comment by rhine3** [comment]
In Raven tutorial:
* Fix formatting of parameters itemized list under the Split Raven annotations and audio files section
* Fix typo: min_label_len=1 vs. "contain at least 0.5s of label"

**Comment by sammlapp** [comment]
Add documentation for how to use model.metrics

**Comment by rhine3** [comment]
^ In addition to that documentation write a whole tutorial about interpretation of results: precision/recall, histograms

**Comment by rhine3** [comment]
Additional suggestions from Lydia:

* In general: flag certain common machine learning issues/things to be aware of, and then provide links to other information sources to point any beginners in the right direction?
* Prepare audio data – would be good to include a few points about what this should look like (i.e. sound clips should all be same length), and especially the issue of balanced data (when I first attempted machine learning I had no idea I had to balance the classes)
* Training model – would be good to show how to plot validation loss alongside training loss, and other metrics of interest, and maybe include a link to a website/info about inspecting for over and underfitting etc.  
* Plotting metrics – e.g. precision recall curve, ROC etc.,
* What to do next? – I found it relatively easy to train many not so great models just by trial and error– would be really great to have info for people on how to really understand if they have trained a useful model, how to interpret results etc. Especially how to use the model.
* Full example from the lab: get a feel for what you actually do when training an ML model – e.g. the typical size of your training set, the processes you did to assemble it (e.g. over and under sampling), typical number of epochs train for, the steps you go through to ensure you have a good model. We could also do this as a blog post if it's not appropriate for the manual.

**Comment by rhine3** [comment]
More ideas for general machine learning concepts to add to docs:

* Sam's glossary
* Sam's OPSO basics notebook
* Sam's links:
  * https://jalammar.github.io/visual-interactive-guide-basics-neural-networks/
  * https://jalammar.github.io/feedforward-neural-networks-visual-interactive/
  * https://victorzhou.com/blog/intro-to-neural-networks/
  * https://victorzhou.com/blog/intro-to-cnns-part-1/
* What are the conceptual differences between training and validation?
* Why do augmentation on the validation data?
* Validation set might give an inflated estimate of generalizability/performance if training and validation data came from same dataset (e.g. same field site) and especially from same source files (e.g. a training sample from 0,5s and a validation sample from 5,10s of the same file). Could avoid this, e.g. using temporal partitioning to avoid including validation files from the same training data
* Why batch size matters, especially with batch-normalizing algorithms
* Metrics for memory/processor usage and why they matter (e.g. batch size)

**Comment by sammlapp** [comment]
I'd rather these be separated out into individual issues so that we can close them when they are compelted

**Comment by louisfh** [comment]
I am going to close this and open separate issues for the unresolved points

I believe the below are resolved/no longer relevant:
- Difference between model.load() and Class.from_checkpoint():
- How to cite OpSo
- Raven tutorial formatting.

Issues that I will open:
- Purpose of Resnet18Binary and Resnet18Multiclass?
- Tutorial/instructions for what to do once you have a trained model: interpreting model performance metrics, precision-recall/ROC/histograms,  validation vs training metrics.
- Documentation (blog post? Tutorial notebook?) explaining ML concepts such as augmentation, effects of batch size, memory usage etc.
- Sam's suggestion for showing how a custom action can be used in a pipeline.
